libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -146,16 +146,21 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
         val salesforceSDKManager = SalesforceSDKManager.getInstance()
 
         /*
-         * For Salesforce Identity API UI Bridge support, the overriding front door bridge URL to
-         * use in place of the default initial login URL plus the optional web server flow code
-         * verifier accompanying the front door bridge URL
+         * For Salesforce Identity API UI Bridge support, the overriding
+         * frontdoor bridge URL to use in place of the default initial login URL
+         * plus the optional web server flow code verifier accompanying the
+         * frontdoor bridge URL.
          */
-        val frontDoorBridgeUrl = salesforceSDKManager.frontDoorBridgeUrl
-        val frontDoorBridgeCodeVerifier = salesforceSDKManager.frontDoorBridgeCodeVerifier
-        val isUsingFrontDoorBridge = frontDoorBridgeUrl != null
-        // Reset the the Salesforce SDK manager's UI bridge support.
-        salesforceSDKManager.frontDoorBridgeUrl = null
-        salesforceSDKManager.frontDoorBridgeCodeVerifier = null
+        val isUsingFrontDoorBridge = isFrontdoorBridgeUrlIntent(intent) || isQrCodeLoginUrlIntent(intent)
+        val uiBridgeApiParameters = when (isQrCodeLoginUrlIntent(intent)) {
+            true -> uiBridgeApiParametersFromQrCodeLoginUrl(intent.data?.toString())
+            else -> intent.getStringExtra(EXTRA_KEY_FRONTDOOR_BRIDGE_URL)?.let { frontdoorBridgeUrl ->
```

Comment:
NIT:  [Kotlin style guide says to prefer if to when for binary conditions](https://kotlinlang.org/docs/coding-conventions.html#if-versus-when).

Reply:
I gotcha in 👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2628/commits/e2f40be4bff9149bfb5be9bde9694868cfa7d287